File: split/printdelimiter.pas
procedure printdelimiter(p: halfword);
var
  a: integer;
begin
  a := mem[p].qqqq.b0 * 256 + mem[p].qqqq.b1 - 0;
  a := a * 4096 + mem[p].qqqq.b2 * 256 + mem[p].qqqq.b3 - 0;
  if a < 0 then
    printint(a)
  else
    printhex(a);
end;

File: cpp/printdelimiter.cpp
#include "printdelimiter.h"
#include "printint.h"
#include "printhex.h"

void printdelimiter(halfword p)
{
	int a = mem[p].qqqq.b0*0x1'00+mem[p].qqqq.b1;
	a = a*0x10'00+mem[p].qqqq.b2*0x1'00+mem[p].qqqq.b3;
	if (a < 0)
		printint(a);
	else
		printhex(a);
}
